iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -176,10 +179,27 @@ public boolean shouldSkipCombine(Path path, Configuration conf) {
   }
 
   @Override
-  public VectorizedSupport.Support[] getSupportedFeatures() {
+  public VectorizedSupport.Support[] getSupportedFeatures(HiveConf hiveConf, Set<String> tableNames) {
     // disabling VectorizedSupport.Support.DECIMAL_64 as Parquet doesn't support it, and we have no way of telling
     // beforehand what kind of file format we're going to hit later
-    return new VectorizedSupport.Support[]{  };
+    boolean onlyOrcFiles = true;
+    try {
+      Hive hiveDb = Hive.get(hiveConf, false);
```

Comment:
Is it possible to make this decision in Vectorizer::getVectorizedInputFormatSupports? Basically trying to avoid the mixup of HMS related information in InputFormats, as they will be used in AppMasters and tasks. I agree that getSupportedFeatures will be called mainly during compilation, but it will be good to keep HMS interactions outside of inputformats.

Reply:
Thanx Rajesh, I have changed it. Just a bit of hardcoding required for the iceberg related stuff, since hive-exec doesn't have iceberg dependency

Reply:
I agree that doing HMS calls from InputFormat is not a nice pattern, but I'm not really fond of hardcoding Iceberg related stuff to hive-exec either.

Can't we refactor Vectorizer instead so that supplies the Table object or its properties to getSupportedFeatures()? Perhaps we could get this info from `verifyAndSetVectorPartDesc` through `addVectorizedInputFileFormatSupport` and finally through `getVectorizedInputFormatSupports` ?

It's a bigger refactor but the responsibility of deciding decimal64 or other vector features would remain solely within InputFormats.